sycl/source/detail/config.cpp

```diff
@@ -161,30 +161,25 @@ void dumpConfig() {
 #undef CONFIG
 }
 
-// Array is used by SYCL_DEVICE_FILTER and SYCL_DEVICE_ALLOWLIST and
-// ONEAPI_DEVICE_SELECTOR
-const std::array<std::pair<std::string, info::device_type>, 6> &
+// Array is used by SYCL_DEVICE_ALLOWLIST and ONEAPI_DEVICE_SELECTOR
+const std::array<std::pair<std::string, info::device_type>, 5> &
 getSyclDeviceTypeMap() {
-  static const std::array<std::pair<std::string, info::device_type>, 6>
-      SyclDeviceTypeMap = {{{"host", info::device_type::host},
```

Comment:
The [sycl_ext_oneapi_filter_selector](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_filter_selector.asciidoc) still accepts "host" as device type and backend, so we can't remove it from here just yet.

Reply:
can we just drop that from both? My understanding that even if the "host" is part of the extension, it cannot be selected because the support for the device itself has been dropped.

Reply:
Problem is that the extension is supported, so we normally can't just change it out of the blue. Removing it might mean that `filter_selector` would start failing at construction with "host" input rather than on device selection.

@gmlueck | @jbrodman - Thoughts on this? We should probably write a new variant of this extension using SYCL 2020 device selector structure.

Reply:
I took back the host in both device type and backend. The "deprecate/remove sycl_ext_oneapi_filter_selector" job can be done separately from this environment variable removal.